File: javascript/1760.Minimum-Limit-of-Balls-in-a-Bag.js
/**
 * @param {number[]} nums
 * @param {number} maxOperations
 * @return {number}
 */
var minimumSize = function (nums, maxOperations) {
  const canDevide = (maxBalls) => {
    let ops = 0;

    for (const num of nums) {
      ops += Math.ceil(num / maxBalls) - 1;

      if (ops > maxOperations) {
        return false;
      }
    }

    return true;
  };

  for (let i = 1; i < Math.max(...nums); i++) {
    if (canDevide(i)) {
      return i;
    }
  }
};

File: python/1760.Minimum-Limit-of-Balls-in-a-Bag.py
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        def canDivide(maxBalls):
            ops = 0

            for num in nums:
                ops += ceil(num / maxBalls) - 1
                
                if ops > maxOperations:
                    return False
            
            return True

        left = 1
        right = max(nums)
        ans = right

        while left <= right:
            mid = left + (right - left) // 2
            if canDivide(mid):
                ans = mid
                right = mid - 1
            else:
                left = mid + 1

        return ans